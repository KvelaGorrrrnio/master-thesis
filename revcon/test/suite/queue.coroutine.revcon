@coroutine

coroutine main(queue int q) {
  local int n

  # E1
  n += 2
  enqueue n q

  # E2
  n += 8
  enqueue n q

  # E3
  n -= 13
  enqueue n q

  # D1
  dequeue n q
  n -= 2

  # D2
  dequeue n q
  n -= 8

  delocal int n
}
