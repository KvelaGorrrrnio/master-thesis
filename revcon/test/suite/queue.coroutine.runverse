test/suite/queue.coroutine.revcon:
[store]
  main: Ã˜
  q: [>>]

[stack]
[spawn main main end () | unresume main (q) | despawn main main begin ()] <|> []

