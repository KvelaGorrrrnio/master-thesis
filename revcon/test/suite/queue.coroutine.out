test/suite/queue.coroutine.revcon:
[store]
  main: Ã˜
  q: [>-13>]

[stack]
[spawn main main begin () | resume main (q) | despawn main main end ()] <|> []

